### PR TITLE
CP-12112: Allow new service providers

### PR DESCRIPTION
**Pull request by ruijialin-avalabs**
## Description

**Ticket: [CP-12112]**

Please provide:

- A summary of the changes
- Motivation and context for this change
- Any dependencies introduced
- (If breaking) migration steps and instructions

- make service provider and payment method zod schema as string to allow any new provider or payment method
- fix NaN balance in TokenList if the balance is over 1000

## Screenshots/Videos

Include relevant screenshots or screen recordings of iOS and Android.

https://github.com/user-attachments/assets/a83f246a-75da-436a-b298-db25a9ccb6a5

https://github.com/user-attachments/assets/a5b29735-8fb2-42ab-9a98-a07dada7d581

## Testing

**Dev Testing (if applicable)**

- Provide steps to test the happy path of your feature
- Provide steps to test edge cases and error states
- Trigger a build on bitrise and reference it here
- Move the ticket into the "Testing" column on Jira

For dev testing, you can disable some of the service providers and payment methods, and it should still show and work the same as before.  the text and logo might be slightly different, since we use everything lcoally, but if they are not mapped, we use logo and name from the api response

**QA Testing (if applicable)**

- Provide instructions for QA to test this feature thoroughly
- State expected behavior / acceptance criteria

## Checklist

Please check all that apply (if applicable)

- [ ] I have performed a self-review of my code
- [ ] I have verified the code works
- [ ] I have included screenshots / videos of android and ios
- [ ] I have added testing steps
- [ ] I have added/updated necessary unit tests
- [ ] I have updated the documentation

### CODE REVIEW

File: packages/core-mobile/app/new/features/meld/components/TokenList.tsx [hunk 1]
```diff
@@ -171,7 +171,12 @@ export const TokenList = ({
               </Text>
               <View sx={{ flexDirection: 'row' }}>
                 <SubTextNumber
-                  number={Number(item.tokenWithBalance.balanceDisplayValue)}
+                  number={Number(
+                    item.tokenWithBalance.balanceDisplayValue.replaceAll(
+                      ',',
+                      ''
+                    )
```
Comment (atn4z7) [inline packages/core-mobile/app/new/features/meld/components/TokenList.tsx:178]: this will probably happen again unless we do it in SubTextNumber

File: packages/core-mobile/app/new/features/meld/components/PaymentMethodIcon.tsx [hunk 2]
```diff
@@ -37,7 +39,7 @@ export const PaymentMethodIcon = ({
   ) : (
     <Image
       accessibilityRole="image"
-      sx={{ width: size, height: size }}
+      sx={{ width: size, height: size, borderRadius: 1000 }}
```
Comment (B0Y3R-AVA) [inline packages/core-mobile/app/new/features/meld/components/PaymentMethodIcon.tsx:42]: border radius 1000 seems crazy? worth leaving an inline comment here why we'd need it so high?

File: packages/core-mobile/app/new/features/meld/hooks/useSelectAmount.tsx [hunk 3]
```diff
@@ -13,6 +13,7 @@ import { useFormatCurrency } from 'common/hooks/useFormatCurrency'
 import { useNavigation } from '@react-navigation/native'
 import { ACTIONS } from 'contexts/DeeplinkContext/types'
 import { useDebouncedCallback } from 'use-debounce'
+import { capitalize } from 'lodash'
```
Comment (atn4z7) [inline packages/core-mobile/app/new/features/meld/hooks/useSelectAmount.tsx:16]: let's reuse the existing util
